### PR TITLE
🔧(frontend) use new jsx transform

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose

Since React 17, a new JSX transformer is available.
It does not require anymore to load React package from jsx files and can in some case optimize bundle size.

## Proposal

- [x] Update babel config to use new JSX transform
- [x] Run `npx react-codemod update-react-imports` to update codebase

### CODE REVIEW

File: src/frontend/js/components/CourseRunEnrollment/index.tsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
-import React, { useCallback, useEffect, useReducer } from 'react';
+import { useCallback, useEffect, useReducer } from 'react';
+import * as React from 'react';
```
Comment (mbenadda) [inline src/frontend/js/components/CourseRunEnrollment/index.tsx:2]: This seems worse than before: 2 lines and a wildcard import. Can we just leave it like it was?

Comment (jbpenrath) [reply src/frontend/js/components/CourseRunEnrollment/index.tsx:2]: Oh yes ! I trusted `react-codemod`...

Comment (mbenadda) [reply src/frontend/js/components/CourseRunEnrollment/index.tsx:2]: IMO we can leave `import React from 'react'` wherever it inserted `import * as React`. What do you think?

Comment (jbpenrath) [reply src/frontend/js/components/CourseRunEnrollment/index.tsx]: Yes I also find the syntax better. But I took the opportunity to import only stuff used when there is only one thing used from React.

Comment (mbenadda) [reply src/frontend/js/components/CourseRunEnrollment/index.tsx]: No more React Football Club!

**Review by mbenadda — APPROVED** [review]

**Pull request merged**